frontends/p4/toP4/toP4.cpp

```diff
@@ -46,17 +46,14 @@ void ToP4::end_apply(const IR::Node *) {
               "inconsistent vectorSeparator");
 }
 
-// Try to guess whether a file is a "system" file
-bool ToP4::isSystemFile(cstring file) {
-    if (noIncludes) return false;
-    if (file.startsWith(p4includePath)) return true;
-    return false;
-}
-
 cstring ToP4::ifSystemFile(const IR::Node *node) {
```

Comment:
Since you are touching it, maybe change it to return `std::optional<std::string>` (or `std::optional<cstring>`? The null-distinct-from-empty behavior of `cstring` is a bit anoying and bad design in my opinion. I'm not sure how big impact this would have though...